onnxruntime/test/optimizer/transpose_optimizer_test.cc

```diff
@@ -3502,118 +3502,150 @@ TEST(TransposeOptimizerTests, TestWhere) {
+      auto* input2_arg = MakeInput<uint8_t>(builder, {std::vector<int64_t>{}}, std::vector<int64_t>{}, {10});
+      auto* transpose_1_out_0 = builder.MakeIntermediate();
+      auto* quantizelinear_1_out_0 = builder.MakeIntermediate();
+      auto* transpose_2_out_0 = builder.MakeOutput();
+
+      auto& transpose_1 = builder.AddNode("Transpose", {input0_arg}, {transpose_1_out_0});
+      transpose_1.AddAttribute("perm", std::vector<int64_t>{0, 3, 1, 2});
+      builder.AddNode("QuantizeLinear", {transpose_1_out_0, input1_arg, input2_arg}, {quantizelinear_1_out_0},
+                      q_domain);
+      auto& transpose_2 = builder.AddNode("Transpose", {quantizelinear_1_out_0}, {transpose_2_out_0});
+      transpose_2.AddAttribute("perm", std::vector<int64_t>{0, 2, 3, 1});
+    };
 
-    auto& transpose_1 = builder.AddNode("Transpose", {input0_arg}, {transpose_1_out_0});
-    transpose_1.AddAttribute("perm", std::vector<int64_t>{0, 3, 1, 2});
-    builder.AddNode("QuantizeLinear", {transpose_1_out_0, input1_arg, input2_arg}, {quantizelinear_1_out_0});
-    auto& transpose_2 = builder.AddNode("Transpose", {quantizelinear_1_out_0}, {transpose_2_out_0});
-    transpose_2.AddAttribute("perm", std::vector<int64_t>{0, 2, 3, 1});
-  };
+    auto check_optimized_graph_1 = [&](InferenceSessionWrapper& session) {
+      int transpose_cost = EstimateTransposeCost(session.GetGraph());
+      EXPECT_EQ(transpose_cost, 0);
+    };
 
-  auto check_optimized_graph_1 = [&](InferenceSessionWrapper& session) {
-    int transpose_cost = EstimateTransposeCost(session.GetGraph());
-    EXPECT_EQ(transpose_cost, 0);
+    TransformerTester(build_test_case_1,
+                      check_optimized_graph_1,
+                      TransformerLevel::Default,
+                      TransformerLevel::Level1,
+                      /*opset_version*/ {15, 18});
   };
 
-  TransformerTester(build_test_case_1,
-                    check_optimized_graph_1,
-                    TransformerLevel::Default,
-                    TransformerLevel::Level1,
-                    /*opset_version*/ {15, 18});
+  test_case();
+#if !defined(DISABLE_CONTRIB_OPS)
+  test_case(kMSDomain);  // Use com.microsoft.QuantizeLinear
+#endif
 }
 
 TEST(TransposeOptimizerTests, TestQuantizeLinearScalarIgnoreAxis) {
-  auto build_test_case_1 = [&](ModelTestBuilder& builder) {
-    auto* input0_arg = MakeInput<float>(builder, {{2, -1, 6, 3}}, {2, 4, 6, 3}, 0.0, 1.0);
-    auto* input1_arg = MakeInput<float>(builder, {std::vector<int64_t>{}}, std::vector<int64_t>{}, {2.3f});
-    auto* input2_arg = MakeInput<uint8_t>(builder, {std::vector<int64_t>{}}, std::vector<int64_t>{}, {10});
-    auto* transpose_1_out_0 = builder.MakeIntermediate();
-    auto* quantizelinear_1_out_0 = builder.MakeIntermediate();
-    auto* transpose_2_out_0 = builder.MakeOutput();
+  auto test_case = [&](const std::string q_domain = "") {
```

Comment:
nit: there are `const std::string` as well as `const std::string&` parameters in these changes. should we consistently use the latter?

Reply:
Ah yeah, I'll fix those. That was not intentional. ty

Reply:
Fixed